From c04282be56fd55a555f747a5753cf7425fb233ff Mon Sep 17 00:00:00 2001
From: "pj.chen" <chen.pj@inventec.com>
Date: Tue, 1 Mar 2022 07:10:34 +0000
Subject: [PATCH 14/27] PSUSensor - add extra driver config

Symptom/Reason:
    Add extra driver config

Root Cause:
    N/A

Solution/Change:
    [PSUSensor]
    - add pac1934/2/1 driver config
    - add SBRMI driver config

Entry Test:

ipmitool sdr elist
PSU1_VIN         | 70h | ok  | 10.0 | 216.80 Volts
PSU1_VOUT        | 71h | ok  | 10.1 | 12.10 Volts
PSU1_Input_Power | 72h | ok  | 10.2 | 315.40 Watts
PSU1_Out_Power   | 73h | ok  | 10.3 | 283.50 Watts
PSU1_IIN         | 79h | ok  | 10.4 | 1.47 Amps
PSU1_IOUT        | 7Ah | ok  | 10.5 | 23.92 Amps
PSU1_Fan1        | 7Bh | ok  | 10.6 | 6254 RPM
PSU1_Temp1       | 7Ch | lnc | 10.7 | 25.14 degrees C
PSU1_Temp2       | 7Dh | ok  | 10.8 | 35.18 degrees C
PSU1_Temp3       | 7Eh | ok  | 10.9 | 42.74 degrees C
Pwm_PSU1_Fan_1   | 7Fh | ok  | 10.10 | 30.18 unspecifi
FAN0_V           | 86h | ok  | 29.12 | 12.19 Volts
FAN1_V           | 87h | ok  | 29.13 | 12.19 Volts
FAN2_V           | 88h | ok  | 29.14 | 12.19 Volts
FAN3_V           | 89h | ok  | 29.15 | 12.19 Volts
FAN4_V           | 8Ah | ok  | 29.16 | 12.19 Volts
FAN5_V           | 8Bh | ok  | 29.17 | 12.19 Volts
FAN0_Power       | 76h | ok  | 29.18 | 0 Watts
FAN1_Power       | 77h | ok  | 29.19 | 0 Watts
FAN2_Power       | 78h | ok  | 29.20 | 0 Watts
FAN3_Power       | 79h | ok  | 29.21 | 0 Watts
FAN4_Power       | 98h | ok  | 29.22 | 0 Watts
FAN5_Power       | 99h | ok  | 29.23 | 0 Watts

[comment for 2.12 upgrade]
some power config in propertyInitialize is already set, remove code in this patch

---
 src/PSUSensorMain.cpp | 12 ++++++++----
 1 file changed, 8 insertions(+), 4 deletions(-)

diff --git a/src/PSUSensorMain.cpp b/src/PSUSensorMain.cpp
index d025773..bfbf94a 100644
--- a/src/PSUSensorMain.cpp
+++ b/src/PSUSensorMain.cpp
@@ -44,6 +44,10 @@ static constexpr bool debug = false;
 
 static const I2CDeviceTypeMap sensorTypes{
     {"VIRTUAL_PSU", I2CDeviceType{"virtual_psu", true}},
+    {"PAC1934", I2CDeviceType{"pac1934", true}},
+    {"PAC1932", I2CDeviceType{"pac1932", true}},
+    {"PAC1931", I2CDeviceType{"pac1931", true}},
+    {"SBRMI", I2CDeviceType{"sbrmi", true}},
     {"ADM1266", I2CDeviceType{"adm1266", true}},
     {"ADM1272", I2CDeviceType{"adm1272", true}},
     {"ADM1275", I2CDeviceType{"adm1275", true}},
@@ -1033,10 +1037,10 @@ void propertyInitialize(void)
         {"vout32", PSUProperty("Output Voltage", 255, 0, 3, 0)},
         {"vmon", PSUProperty("Auxiliary Input Voltage", 255, 0, 3, 0)},
         {"in0", PSUProperty("Output Voltage", 255, 0, 3, 0)},
-        {"in1", PSUProperty("Output Voltage", 255, 0, 3, 0)},
-        {"in2", PSUProperty("Output Voltage", 255, 0, 3, 0)},
-        {"in3", PSUProperty("Output Voltage", 255, 0, 3, 0)},
-        {"in4", PSUProperty("Output Voltage", 255, 0, 3, 0)},
+        {"in1", PSUProperty("Input Current", 255, 0, 3, 0)},
+        {"in2", PSUProperty("Input Current", 255, 0, 3, 0)},
+        {"in3", PSUProperty("Input Current", 255, 0, 3, 0)},
+        {"in4", PSUProperty("Input Current", 255, 0, 3, 0)},
         {"in5", PSUProperty("Output Voltage", 255, 0, 3, 0)},
         {"in6", PSUProperty("Output Voltage", 255, 0, 3, 0)},
         {"in7", PSUProperty("Output Voltage", 255, 0, 3, 0)},
-- 
2.25.1